env_spec.py

```diff
@@ -1,53 +1,82 @@
 import re
 
-
-def check_input(field_array):
+valid_types_array = [
+    "color",
+    "date",
+    "datetime-local",
+    "email",
+    "month",
+    "number",
+    "password",
+    "tel",
+    "text",
+    "time",
+    "url",
+    "week",
+]
+
+
+def check_input(field_type_array):
     """
     Checks syntax of input(uppercase latin chars,numeric digits, underscores).
     Returns False for error or True for correct syntax.
     """
     alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
 
-    for line in field_array:
-        ret = re.match(alphanumeric_that_does_not_start_with_digit, line)
+    for line in field_type_array:
+        ret = re.match(alphanumeric_that_does_not_start_with_digit, line[0])
 
         if ret is None:
-            # print("SYNTAX ERROR:WRONG SYMBOLS!")
             return False
 
     return True
 
 
-def render_env_var_spec(env_var_name):
+def check_type(input_array):
+    """
+    checks if given type is valid and returns True, else returs False
+    """
+    for type in input_array:
```

Comment:
Can you rename `type` to something else (e.g. `kind` or `_type`) [since it's already used by Python](https://docs.python.org/3/library/functions.html#type)? It will help avoid future confusion.